rails_programming/forms_and_authentication/form_basics.md

```diff
@@ -273,3 +265,15 @@ At this point, you should have a solid understanding of how forms work in genera
 
 ### Additional Resources
 This section contains helpful links to other content. It isn't required, so consider it supplemental.
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/form-basics#railsifying-your-form">What is a CSRF Token and why is it necessary?</a>
+* <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/form-basics#making-forms-into-params">What is the `name` attribute of a form input element and what does it do?</a>
+* <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/form-basics#making-forms-into-params">How do you nest attributes under a single hash in `params`?</a>
+* <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/form-basics#form-helpers-formwith">When using the `form_with` helper, what does Rails set inputs' IDs to?</a>
```

Comment:
I think I'd drop this question and add one about form_with and models, because IMO that's a more core topic of the lesson.

Reply:
I had this one that I discarded previously: How do you pass `form_with` a model object?

Reply:
That one seems good to me!